blockchain/projects.go

```diff
@@ -187,92 +157,104 @@ func (p *Projects) CreateInitialAccounts(projectID uuid.UUID) ([]*model.Account,
 		if err != nil {
 			return nil, err
 		}
-
-		account := model.AccountFromFlow(flowAccount, projectID)
-		account.Index = i
-		account.ID = uuid.New()
-		accounts[i] = account
 	}
 
-	return accounts, nil
+	return &p.accountsNumber, nil
 }
 
 // CreateAccount creates a new account and return the account model as well as record the execution.
-func (p *Projects) CreateAccount(projectID uuid.UUID) (*model.Account, error) {
+func (p *Projects) CreateAccount(projectID uuid.UUID) error {
 	p.mutex.load(projectID).Lock()
 	defer p.mutex.remove(projectID).Unlock()
 	em, err := p.load(projectID)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	account, tx, result, err := em.createAccount()
+	_, tx, result, err := em.createAccount()
 	if err != nil {
-		return nil, err
+		return err
 	}
 
 	exe := model.TransactionExecutionFromFlow(projectID, result, tx)
 	err = p.store.InsertTransactionExecution(exe)
 	if err != nil {
-		return nil, err
+		return err
 	}
 
-	return model.AccountFromFlow(account, projectID), nil
+	return nil
 }
 
 // DeployContract deploys a new contract to the provided address and return the updated account as well as record the execution.
 func (p *Projects) DeployContract(
 	projectID uuid.UUID,
 	address model.Address,
 	script string,
-) (*model.Account, error) {
+) (*model.ContractDeployment, error) {
 	p.mutex.load(projectID).Lock()
 	defer p.mutex.remove(projectID).Unlock()
 	em, err := p.load(projectID)
 	if err != nil {
 		return nil, err
 	}
 
-	result, tx, err := em.deployContract(address.ToFlowAddress(), script)
+	contractName, err := GetContractName(script)
 	if err != nil {
 		return nil, err
 	}
-	if result.Error != nil {
-		return nil, result.Error
-	}
 
-	exe := model.TransactionExecutionFromFlow(projectID, result, tx)
-	err = p.store.InsertTransactionExecution(exe)
+	flowAccount, _, err := em.getAccount(address.ToFlowAddress())
 	if err != nil {
 		return nil, err
 	}
+	if _, ok := flowAccount.Contracts[*contractName]; ok {
```

Comment:
We should handle this differently going forward. I say we don't reset the state anymore. If a user wants to redeploy the same contract that's already on an account we just remove that contract and deploy it again, we don't clear all other accounts and the state.